lib/resty/rediscluster.lua

```diff
@@ -412,13 +413,13 @@ end
 local function handle_command_with_retry(self, target_ip, target_port, asking, cmd, key, ...)
     local config = self.config
 
-    key = tostring(key)
-    local slot = redis_slot(key)
+    -- key will be passed to resty.redis
+    local slot = redis_slot(tostring(key))
 
     for k = 1, config.max_redirection or DEFAULT_MAX_REDIRECTION do
 
         if k > 1 then
-            ngx.log(ngx.NOTICE, "handle retry attempts:" .. k .. " for cmd:" .. cmd .. " key:" .. key)
+            ngx.log(ngx.NOTICE, "handle retry attempts:" .. k .. " for cmd:" .. cmd .. " key:" .. pl_pretty.write(key))
```

Comment:
```suggestion
            ngx.log(ngx.NOTICE, "handle retry attempts:" .. k .. " for cmd:" .. cmd .. " key:"
                    .. type(key) == "table" and pl_pretty.write(key) or key)
```

Reply:
^ sorry I didn't realize that `pl` is not a dependency that is included in lua-resty-cluster's rockspec.
It should be okay that only log the table ref with `tostring`.

```suggestion
            ngx.log(ngx.NOTICE, "handle retry attempts:" .. k .. " for cmd:" .. cmd .. " key:" .. tostring(key))
```